src/JSInterop/Microsoft.JSInterop.JS/src/src/Microsoft.JSInterop.ts

```diff
@@ -237,6 +231,8 @@ module DotNet {
     });
 
     if (result instanceof Function) {
+      result = result.bind(lastSegmentValue);
+      cachedJSFunctions[identifier] = result;
```

Comment:
Previously we also had a bug where it never cached the lookup results so it always recreated them on each call.

Reply:
Is it reasonable to have a test for this?

Reply:
I'll see if this is a reasonable thing to add to JS interop E2E tests (though of course they are in a different repo).

Reply:
That's fine, I think we still want to know if this breaks.